src/command_executor.cc

```diff
@@ -128,6 +131,7 @@ int CommandExecutor::Execute(const std::string &line) {
       if (!pipe_fds_list.empty()) {
         built_In_Command_ptr = &command_parse_result_list[i];
       } else {
+        close(save_fd);
```

Comment:
Does this necessary? I think line 153 might close fd correctly when we only have one build-in command.